### DISCUSSION TITLE
what happens when opening corrupted pdf document in sumatra?

### DISCUSSION THREAD

**Discussion by ghost** [sumatrapdfreader/sumatrapdf#2517] [Q&A]
I have hundreds of pdf and epub files in my collection, and as you all know, flash drives such as USBs and SSDs are not ideal for long term storage. But personally I have reason to stick with this for some time.

So there's good chance those files get corrupted if the device is off power for long enough.

And I don't have time and energy to thoroughly compare the files with the original word by word, file by file. Oh no, I can't.

So here's the question.
How does sumatra pdf deal with these corrupt files when trying to open it?
Does it refuse to open it compeletely, or opens the file as-is, so I see some part of it broken or missing?

Those files are not in any way critical or contain any confidential data, I just want to know if I can feel safe the files are intact if it opens normally.

I can save the checksum of each file but if sumatra pdf gives me any signs of the file being corrupted, it would be much better :)

Thank you in advance.

**Comment by GitHubRulesOK** [comment]
There is no way to be sure if a pdf is corrupted/broken many actually are and PDF apps will do their own sub set of fixes to cope. The worst cases are a file is truncated for one of many reasons like incomplete download or trailer bad edit by annotation.
 The way a PDF reader generally works is open bytes at start to check it starts `%PDF-` (40 bit signature) or similar with version e.g. %PDF-2.0 then most likely read the end of file length for `%EOF` (but there can be more than one of those) then generally work backwards looking for `startxref` (if it was not at the beginning of web pdf file) the `xref` table can be corrupted for many reasons and the reader tries to work through those decimal addresses looking for the catalog of pages (so index after index after index as font can be too) the file is tree like but adobe built it code named Carousel thus goes round and round.
 A bad disk sector recovered file is a nightmare as there could easily be two file starts and no endings. For assurance in the days pre self healing smart drives there was crc32 or better MD5 for directory of files but clearly that just confirms the file went bad at some time same as attempting open with a bad result.
 For basic recovery try different readers including Acrobat Reader and reprint as a new printout. However SumatraPDF printing converts to image format (not with searchable text fonts)
